src/System.Net.HttpListener/tests/HttpListenerResponseTests.cs

```diff
@@ -13,18 +14,17 @@ namespace System.Net.Tests
     public class HttpListenerResponseTests : IDisposable
     {
         private HttpListenerFactory Factory { get; }
-        private HttpClient Client { get; }
```

Comment:
I actually like the fact we're using HttpClient: we should continue to do so to ensure that the client/server stacks can communicate with themselves. (It's less interesting that a custom test client can achieve the same thing.)

That said, you could duplicate all tests here into another file to the granular control through a Socket and a custom HTTP test server.